src/Agent.Plugins/BuildDrop/BuildDropPlugin.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent;
+using Agent.Sdk;
+
+namespace Agent.Plugins.BuildDrop
+{
+    public abstract class BuildDropTaskPluginBase : IAgentTaskPlugin
+    {
+        public abstract Guid Id { get; }
+        public string Version => "0.138.0"; // Publish and Download tasks will be always on the same version.
+        public string Stage => "main";
+
+        public async Task RunAsync(AgentTaskPluginExecutionContext context, CancellationToken token)
+        {
+            ArgUtil.NotNull(context, nameof(context));
+
+            // Artifact Name
+            string artifactName;
+            if (!context.Inputs.TryGetValue(ArtifactEventProperties.ArtifactName, out artifactName) ||
+                string.IsNullOrEmpty(artifactName))
+            {
+                throw new Exception(StringUtil.Loc("ArtifactNameRequired"));
+            }
+
+            // Path
+            string targetPath;
```

Comment:
Please leave a comment here for me.
TODO: Translate targetPath from container to host (Ting)

Reply:
added